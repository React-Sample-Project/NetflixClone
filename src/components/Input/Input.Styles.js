import styled from "styled-components";
export const InputMain = styled.input`
  ${({ customStyle }) => customStyle}
`;
