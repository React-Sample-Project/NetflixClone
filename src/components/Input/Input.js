import React, { forwardRef } from "react";
import { InputMain } from "./Input.Styles.js";

const Input = forwardRef(({ type, ...otherProps }, ref) => (
  <InputMain type={type || "text"} ref={ref} {...otherProps} />
));

export default Input;
