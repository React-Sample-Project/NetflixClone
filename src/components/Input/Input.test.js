import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Input from "./Input";
describe("Testing Input Component", () => {
  it("renders the component", () => {
    const { container } = render(<Input type="password" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("triggers events", () => {
    const onChange = jest.fn();
    const onFocus = jest.fn();
    const onBlur = jest.fn();
    render(<Input onChange={onChange} onFocus={onFocus} onBlur={onBlur} />);
    const input = screen.getByRole("textbox");
    fireEvent.change(input, {
      target: { value: "Testing Jest" },
    });
    fireEvent.focus(input);
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onFocus).toHaveBeenCalledTimes(1);
    fireEvent.blur(input);
    expect(onBlur).toHaveBeenCalledTimes(1);
  });
});
