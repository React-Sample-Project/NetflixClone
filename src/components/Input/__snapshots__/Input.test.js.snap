// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input Component renders the component 1`] = `
<input
  class="sc-AxjAm hyNcgC"
  type="password"
/>
`;
